Source/WebCore/platform/audio/glib/MediaSessionGLib.cpp

```diff
@@ -284,9 +284,13 @@ void MediaSessionGLib::updateNowPlaying(NowPlayingInfo& nowPlayingInfo)
 
     GVariantBuilder propertiesBuilder;
     g_variant_builder_init(&propertiesBuilder, G_VARIANT_TYPE("a{sv}"));
-    g_variant_builder_add(&propertiesBuilder, "{sv}", "Metadata", getMetadataAsGVariant(nowPlayingInfo));
-    emitPropertiesChanged(g_variant_new("(sa{sv}as)", DBUS_MPRIS_PLAYER_INTERFACE, &propertiesBuilder, nullptr));
+
+    auto metadata = getMetadataAsGVariant(nowPlayingInfo);
+    g_variant_builder_add(&propertiesBuilder, "{sv}", "Metadata", metadata);
+
+    auto properties = adoptGRef(g_variant_ref_sink(g_variant_new("(sa{sv}as)", DBUS_MPRIS_PLAYER_INTERFACE, &propertiesBuilder, nullptr)));
```

Comment:
I'm not sure if this is the right solution... but it made lsan happier...

Reply:
Well the problem here was `MediaSessionGLib::emitPropertiesChanged` was accepting a raw pointer to a `GVariant`, so it did not previously take ownership, but both callers were improperly passing ownership. Your changes fix this. Looks good.

That said, `adoptGRef(g_variant_ref_sink())` should be equivalent to doing nothing because the `adoptGRef` just prevents the `GRefPtr` from doing the `g_variant_ref_sink()` itself. Arguably it's better to write it out explicitly like you do here, because using floating refs with `GRefPtr` is pretty confusing and it sure does look like a leak if you don't. I would be quite tempted to change `GRefPtr` to assert whenever an object has a floating ref to force ourselves to sink the floating ref explicitly. But that would be a pretty big change. Anyway, the equivalent shorter way to write this (which we're probably expected to write):

```suggestion
    auto properties = g_variant_new("(sa{sv}as)", DBUS_MPRIS_PLAYER_INTERFACE, &propertiesBuilder, nullptr);
```

I think @calvaris and @carlosgcampos both have opinions on how to use of floating refs with `GRefPtr`, so we should see what they think.

Reply:
If I didn't to the `ref_sink` I would get critical warnings. So yeah it's more verbose but correct indeed.

What you suggest (iiuc, passing the raw pointer to the function accepting a `RefPtr<T>&&`) also works but I worry it looks a bit too magic.

Reply:
> If I didn't to the ref_sink I would get critical warnings.

Sorry, I meant to suggest `GRefPtr properties` but I see I actually suggested `auto properties`. Don't do that. :) That's the worst option, and it also violates the rule that we don't hold ownership in raw pointers.

Reply:
Anyway, what you have now is fine. Possibly slightly confusing since the RefPtr gets constructed implicitly, but it looks correct to me.

Reply:
Agreed, too much implicit stuff going on, I'll rework this tomorrow, thanks for the review!